### ISSUE TITLE
Listener of a component inside router view

### ISSUE THREAD

**Issue by PROgram52bc** [probil/vue-socket.io-extended#408]
I have a component like the following that listens for time sent from the server every second.
```vue
<template>
	<div>
		<h2>Current time: {{ time }}</h2>
	</div>
</template>
<script>
export default {
	sockets: {
		time(m) {
			this.time = m;
			this.$socket.emit('message', `Received time ${m}!`);
		},
	},
	data() {
		return {
			time: 'initial time',
		};
	},
};
</script>
```
This component is rendered with a `<router-view>` component provided by `vue-router`.
It works on the first time switching to the component, like this.
![Screenshot from 2019-07-31 23-04-40](https://user-images.githubusercontent.com/18290763/62223550-b65ee380-b3e7-11e9-906f-625f19c4a37f.png)
but when I switched to some other component and switched back, the time stopped updating, even though the server is still sending the time every second. 
![Screenshot from 2019-07-31 23-06-01](https://user-images.githubusercontent.com/18290763/62223593-cc6ca400-b3e7-11e9-80df-689dae1127b9.png)
Is there any way to fix this? Thanks!

**Comment by probil** [comment]
Hi, @PROgram52bc 👋 
I hope so. Let me investigate it.

If you have time please create a repo or sandbox with a reproduction of the issue
https://codesandbox.io/s/vue

BTW, which version of the library you are using?

**Comment by PROgram52bc** [comment]
Hi @probil,
I am using `vue-socket.io-extended@3.2.2`. 
I will try to recreate the problem for you!

**Comment by PROgram52bc** [comment]
Hi @probil,

I tried creating a sample but ended up finding that very challenging, because it requires the collaboration of a constantly-running public backend server, which I do not have access to...
But I managed to tweak my code a little bit to use a plain `socket.io` instance that makes the connection on the component's `mounted` hook, and it worked even with route switching. The new component looks like this: 
```
<template>
	<div>
		<h2>Current time: {{ time }}</h2>
		<h2>Manual time: {{ manualTime }}</h2>
	</div>
</template>
<script>
import io from 'socket.io-client';
export default {
	sockets: {
		time(m) {
			this.time = m;
			this.$socket.emit('message', `Received time ${m}!`);
		},
	},
	mounted() {
		const s = io('localhost:5000');
		s.on('time', m => {
			this.manualTime = m;
			s.emit('message', `Received manual time!`);
		});
	},
	data() {
		return {
			time: 'initial time',
			manualTime: 'initial time',
		};
	},
};
</script>
```
And after switching to some other component then switching back, I get this output:
![Screenshot from 2019-08-02 20-43-27](https://user-images.githubusercontent.com/18290763/62371006-4460d880-b566-11e9-8ca0-d86f07daec71.png)

So I would suspect there is some sort of trigger you can add to the plugin in order to refresh the connection and continue listening to events sent from the server. 

Let me know if I can help with anything!

**Comment by PROgram52bc** [comment]
Just a quick update. I added this line in my `main.js`:
```
Vue.prototype.$s = io('http://localhost:5000');
```
just so that I have access to `this.$s` in my component, and added the listener in the `mounted` function of my previous component:
```
...
	mounted() {
		this.$s.on('time', m => {
			this.manualTime = m;
			this.$s.emit('message', `Received manual time!`);
		});
	},
...
```
That still worked.
So I reason there is probably something to do with the `socket` option in the component, which should register the listeners every time the component is mounted.

**Comment by probil** [comment]
Thanks @PROgram52bc 
Basically `vue-socket.io-extended` has only one listener for all components and vuex modules.
I wonder why it's not working 🤔 

BTW you don't have to setup everything on codesandbox. You can setup everything locally and give archive or push it as separate repo.

**Comment by PROgram52bc** [comment]
Hi @probil,
I created a minimal example demonstrating the problem.
You can clone the repository to your machine and try it out.
https://github.com/PROgram52bc/vue-socket.io-extended-demo
Let me know if you have any problem running it!

**Comment by probil** [comment]
I will definitely take a look today in the evening.
Thanks @PROgram52bc 👍

**Comment by probil** [comment]
I took a quick look. 
I guess  `v3.2.1` will work in your case. So try to downgrade to exact version.

I going to find a solution that will fix this issue and won't break #400

**Comment by probil** [comment]
@PROgram52bc Fixed in `v4`
